packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -61,7 +61,10 @@ const VirtualizedComboBoxMenu = React.forwardRef(
 
     const surfaceStyles = {
       minWidth,
-      maxWidth: `min(${minWidth * 2}px, 90vw)`,
+
+      // set as constant because we don't want it shifting when items are unmounted
+      maxWidth: minWidth,
```

Comment:
Have you checked how it looks like if virtualization is enabled and item which is not rendered initially is much longer than the width was set before...?